### PR TITLE
Form text input validation

### PR DESCRIPTION
**Pull request by graham-dds**
Addresses https://www.pivotaltracker.com/story/show/170601644

Adds missing text input validation for text input to forms on the frontend and backend.

A schema migration will come in another PR, because that would add noise to an already (somewhat) noisy PR.

Speaking of noise -- There are a lot of single quoute -> double quote changes in `translations.yaml` -- an autoformatting plugin that I have on VSCode keeps `"helping"` me when I edit that file, so I thought I might as well leave the changes in this time if it's all the same.

### CODE REVIEW

File: atst/app.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def make_config(direct_config=None):
         config.read_dict({"default": direct_config})
 
     # Assemble DATABASE_URI value
-    database_uri = "postgres://{}:{}@{}:{}/{}".format(  # pragma: allowlist secret
+    database_uri = "postgresql://{}:{}@{}:{}/{}".format(  # pragma: allowlist secret
```
Comment (graham-dds) [inline atst/app.py]: image

File: atst/forms/validators.py [hunk 2]
```diff
@@ -36,10 +33,11 @@ def _date_range(form, field):
 
 def IsNumber(message=translate("forms.validators.is_number_message")):
```
Comment (leigh-mil) [inline atst/forms/validators.py]: Can we change this to just `Number` to make it consistent with the older validator names?

Comment (graham-dds) [reply atst/forms/validators.py]: Good call!

File: js/lib/input_validations.js [hunk 3]
```diff
@@ -53,6 +73,13 @@ export default {
     unmask: [','],
     validationError: 'Please enter a number',
   },
+  name: {
+    mask: false,
+    match: /[\w \,\.\'\-]{1,100}/,
```
Comment (dandds) [inline js/lib/input_validations.js]: This is sort of a pain, but checking the docs I don't think `\w` means the same thing in Python in JS. In most regex implementations that's used as a character class meaning "any valid alphanumeric character". The question is whether that includes Unicode word characters. It looks like in Python it does by default:

https://docs.python.org/3/library/re.html#regular-expression-syntax

and in JS it does not:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Character_Classes

This means someone would get a frontend validation error for a name with non-latin characters but not a backend validation error. Maybe this answer will help? https://stackoverflow.com/questions/280712/javascript-unicode-regexes

I wouldn't spend too much time on this; if it's a pain to get it right in the JS, I would just loosen that regex for now to allow most characters and document the problem with a PT ticket and a comment there near the validator. We still have the protection of the backend validation; the frontend validation is for UX.

File: translations.yaml [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
+# `{{ "login.title" | translate | safe }}`
 audit_log:
   events:
     default:
-      change: '{from} to {to}'
-      changes: 'Changes:'
-      details: 'Details:'
+      change: "{from} to {to}"
+      changes: "Changes:"
+      details: "Details:"
```
Comment (dandds) [inline translations.yaml]: I don't have a horse in the quote race, but be careful rebasing since there are a lot of changes happening to this file.

**Pull request merged**